rove/src/data_switch.rs

```diff
@@ -103,14 +109,21 @@ impl<'ds> DataSwitch<'ds> {
     // TODO: handle backing sources
     pub async fn get_spatial_data(
         &self,
-        source_id: &str,
+        polygon: Vec<GeoPoint>,
+        data_source: &str,
```

Comment:
changed to be called element, since thats what it is at this point after the splitting...

Reply:
??? Changed what to be element? What this shadows is the name of the `&dyn DataSource` you get out of the map

Reply:
I am confused... I thought you were saying this was incorrect naming?

Reply:
Here you use `data_source` as a name for both the string "frost:air_temp" and the pointer to the frost struct. Do you see? I recommend changing the string to `spatial_id` or some such to avoid the confusing shadowing

Reply:
Or is it just incorrect after I change the name of the underlying variable?

Reply:
in data_switch, it could make sense for data_source to be spatial_id as you mentioned, then propagate that change

Reply:
Yes, that sounds reasonable

Reply:
Is it ok now?

Reply:
Yes, perfect